src/main/java/org/opensearch/security/configuration/ConfigurationRepository.java

```diff
@@ -441,7 +441,7 @@ public Map<CType, SecurityDynamicConfiguration<?>> getConfigurationsFromIndex(Co
             throw new OpenSearchException(e);
         }
 
-        if (logComplianceEvent && auditLog.getComplianceConfig().isEnabled()) {
+        if (logComplianceEvent && auditLog.getComplianceConfig() != null && auditLog.getComplianceConfig().isEnabled()) {
```

Comment:
Added this check here to match this line on main. The integrationTest framework returns this value as null.